src/main/scala/io/qbeast/spark/index/OTreeDataAnalyzer.scala

```diff
@@ -182,35 +180,35 @@ object DoublePassOTreeDataAnalyzer extends OTreeDataAnalyzer with Serializable {
 
   override def analyze(
       dataFrame: DataFrame,
-      oldIndexStatus: IndexStatus,
+      indexStatus: IndexStatus,
```

Comment:
The use of `indexStatus` in the initialization of `IndexStatusChanges`, causes conflicts when we append a new revision. 

When we add new data and the revision changes, the previous `CubeWeights` to merge should be **empty**. In this case, as we use the superseded index status to do the merge in `IndexStatusChanges`, CubeWeights has data on it.  Then we are merging two index structures with different revisions. 

This was the reason for using 

```
// The indexStatus to use
    val indexStatus = spaceChanges match {
      case Some(revisionChange) => IndexStatus(revisionChange.newRevision)
      case None => oldIndexStatus
    }
```

Reply:
But that part is a mistake. You should not create a new revision and pack as a superseded one. The idea of the revision classes is to make clear the life cycle of the meta, and that was a hack. I also saw that in the code, we recreate the new revision or index status multiple times, this can cause many problems. We have to review all the uses of these classes and make sure the way to use them is cristal clear for everybody.